### PR TITLE
add id-query flag "details"

### PR DESCRIPTION
**Pull request by rompetroll**
when flag is set, datahub will inject "datahub_details" as extra property into the resulting entity. this is useful for debugging and understanding how data evolved inside the datahub. "datahub_details" contains information about accessed datasets and change history per dataset.

the implementation is done in a separate module tree, service, and is likely to change when more functionality is moved to service modules. The idea is to replicate functionality in a more decoupled way, while still keeping the same performance levels.

### CODE REVIEW

**Comment by rompetroll** [comment]
closes #169

**Review by ashmanto — APPROVED** [review]

File: internal/server/badger_access.go [hunk 1]
```diff
@@ -23,16 +27,65 @@ type BadgerAccess struct {
...
+	}
+	return scopeArray
+}
+
 func (b BadgerAccess) GetDB() *badger.DB {
 	return b.b
 }
 
-func (b BadgerAccess) LookupDatasetID(datasetName string) (uint32, bool) {
+func (b BadgerAccess) LookupDatasetID(datasetName string) (types.InternalDatasetID, bool) {
```
Comment (gilgamesjh) [inline internal/server/badger_access.go:83]: I like this change :)

File: internal/service/entity/details.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	if err != nil {
+		return nil, err
+	}
+	return details, nil
+}
+
+func (l Lookup) loadDetails(rtxn *badger.Txn, internalEntityId types.InternalID, scope []types.InternalDatasetID) (map[string]interface{}, error) {
+	result := map[string]interface{}{}
+	/*
+		binary.BigEndian.PutUint16(entityIdBuffer, ENTITY_ID_TO_JSON_INDEX_ID)
```
Comment (gilgamesjh) [inline internal/service/entity/details.go:47]: Why are these lines commented out?

File: internal/service/entity/details.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+	badger     store.BadgerStore
+	namespaces namespace.Manager
+}
+
+func NewLookup(s store.BadgerStore) (Lookup, error) {
+	ns := namespace.NewManager(s)
+	return Lookup{s, ns}, nil
+}
+
+func (l Lookup) Details(id string, datasetNames []string) (map[string]interface{}, error) {
```
Comment (gilgamesjh) [inline internal/service/entity/details.go:22]: We should add comments to public API's

**Review by gilgamesjh — APPROVED** [review]

File: internal/service/entity/details.go (hunk 2 above)
Comment (rompetroll) [reply internal/service/entity/details.go:47]: It's just a hint about the key structure we work with

**Pull request merged**